README.md

```diff
@@ -7,11 +7,53 @@ and copying out data at when a user requests it (the [`takeout` function]()).
+
+### How to use these functions in your own project
+
+The `index.js` file has comments about how the functions work; this is about
+how to wire it up.
+
+The developer specifies the paths to data to wipeout or takeout. Those paths live in `user_privacy.json`. The data structures vary for each of the products:
+- For the RTDB, it’s a list of Strings to the path in the database of the form `"</users/uid/follows/..."`
+- For Firestore, it’s a list of Objects of the form `{ "collection": "admins",
+"doc": "UID_VARIABLE", "field": "email" }`.
+- For Storage, it’s a list of Lists with two elements, a bucket name and file name of the form `["cool-project.appspot.com", "users/uid/avatar.jpg"]`
+
+#### Wipeout function
+The wipeout function as written is triggered when a user deletes their account
+using firebase auth, and it performs a wipeout from all three services.
+
+Steps to start using wipeout:
+- [ ] Include the wipeout function and 3 supporting functions (and the requires and convenience variables) in `functions/index.js`
+- [ ] In `user_privacy.json`, add paths to personal information for all the products you’re using.
+- [ ] Make sure you’ve added the ability for the user to delete their account, because that’s what triggers the function
+- [ ] If you’re not using all three products, RTDB, Firestore, and Storage, remove the parts of the function that you don’t need.
+
+As an FYI, the wipeout is implemented by collecting a promise for every deletion event that needs to occur. Only when all promises resolve is the wipeout considered complete. Keep that in mind if you send a confirmation message that wipeout has completed.
+
+#### Takeout function
+The takeout function is triggered via a HTTP request. (The sample app in
+  `/public` has a button that’s wired up to trigger takeout, but in a more
+  traditional app, that would be in settings.)
+
+In order to start using it:
+
+- [ ] Include the function and 4 supporting functions in `functions/index.js.`
+- [ ] In `user_privacy.json`, add:
+  - [ ] A ``"takeoutUploadBucket"`` key that maps to the name of your primary bucket (or in the case of the free tier, your only bucket).
+  - [ ] Paths to personal information for all the products you’re using.
+- [ ] Wire it up in a way appropriate for your platform. For web, I added a
+button in `public/index.html`, that triggers a POST request, and add a rewrite entry in `firebase.json` to trigger the function via HTTP.
+- [ ] If you’re not using all three products, RTDB, Firestore, and Storage,
+remove the parts of the function you don’t need.
+
+Takeout for Storage is weird because we copy the data to a folder for that user, and also create the json of references.
+
+The storage.rules additions are extremely important for takeout. Without them, all the takeout data is freely available. We next the final takeout output under `/takeout`, but check to make sure you don’t have have wildcards in your storage rules, that will also apply to the takeout section.
```

Comment:
Not sure what you're saying with "We next". If this can be a clearer steps (1. add these rules 2. check for wildcards) that might help?

Reply:
Good catch; that looks like editing gone wrong.

Reply:
Replacing with:
> The storage.rules additions are extremely important for takeout; without them, the takeout data is broadly available. By default, the takeout function uploads data to a top level /takeout folder, and the Storage Rules restrict access to the specific user. Be sure to restrict any preexisting rules with wildcards that would grant broader access to takeout data.